Added logic for transformation method (#206)
qbic-pipelines/rnadeseq: assets/RNAseq_report.Rmd

    }
}

# sizeFactors(cds) as indicator of library sequencing depth
if (params$input_type == "featurecounts") {
    sf <- sizeFactors(cds)
} else {
    # If tximport was used, sizeFactors doesn't work, so use workaround as described here: https://support.bioconductor.org/p/97676/
    nm <- assays(cds)[["avgTxLength"]]
    sf <- estimateSizeFactorsForMatrix(counts(cds)/nm)
}
write.table(sf,paste("differential_gene_expression/gene_counts_tables/sizeFactor_libraries.tsv",sep=""), append = FALSE, quote = FALSE, sep = "\t",eol = "\n", na = "NA", dec = ".", row.names = T,  col.names = F, qmethod = c("escape", "double"))

# Write cds assay table to file

> write.table(counts(cds, normalized=T), paste0("differential_gene_expression/gene_counts_tables/deseq2_table.tsv"), append=F, quote = F, sep = "\t", eol = "\n", na = "NA", dec = ".", row.names = T, col.names = T, qmethod = c("escape", "double"))

Reviewer: Could the output be named more explicitly? Like `deseq2_normalized_gene_counts.tsv` ?

Author: Renaming to `deseq2_library_scaled_gene_counts.tsv` to not produce confusion with the rlog/vst normalization stuff that is done later

}

############################## TRANSFORMED AND NORMALIZED COUNTS ###################
# Prepare variables and text snippets for report
vst_cutoff <- 0.05
var_sizeFactors <- var(sf)
run_rlog <- T
trafo_long_text <- "regularized logarithm (rlog)"
trafo_override_text <- ""
nsub_genes_text <- ""
if (params$trafo == "vst-force") {
    run_rlog <- F

>     trafo_long_text <- "variant stabilizing transformation (vst)"

Reviewer: ```suggestion
    trafo_long_text <- "variance stabilizing transformation (vst)"
```

    nsub_genes_text <- paste0("The number of subset genes for the vst transformation was set to ", as.character(params$nsub_genes), ".")
    if (var_sizeFactors >= vst_cutoff) {
        trafo_override_text <- paste0(" This was due to using the `vst-force` argument. It is recommended to rerun the pipeline using the rlog transformation as the size factors of the different samples have a variance of ~", round(var_sizeFactors, 4), ". For high sample variances (i.e. different sequencing depths between samples), the rlog transformation is more suitable [@love2014differential].")
        nsub_genes_text <- ""
    }
} else if (params$trafo == "vst") {
    run_rlog <- F

>     trafo_long_text <- "variant stabilizing transformation (vst)"

Reviewer: ```suggestion
    trafo_long_text <- "variance stabilizing transformation (vst)"
```

<!-- multiQC end -->

# Gene expression data

## Raw and normalized count tables

The raw count table and normalized count tables are available [here](./differential_gene_expression/gene_counts_tables).
The differential expression analysis is performed using the raw gene count table.

> For PCA analysis and heatmap plotting, the `r trafo_long_text` normalized gene counts were employed. 

Reviewer: ```suggestion
For PCA analysis and heatmap plotting, the `r trafo_long_text` normalized gene counts were used/deployed. 
```

        write.table(DE_genes_contrast_genename, file=paste("differential_gene_expression/allgenes/DE_contrast_",contname,".tsv",sep=""), sep="\t", quote=F, col.names = T, row.names = F)

        # Select only significantly DE

>         DE_genes_contrast <- subset(DE_genes_contrast_genename, padj <= params$pval_threshold & abs(log2FoldChange) >= params$logFC_threshold)

Reviewer: Could the parameter `pval_threshold` be renamed to `adj.pval_threshold`?

Author: I changed it to `adj_pval_threshold` just to be on the safe side with dots

>         write.table(DE_genes_contrast_genename, file=paste("differential_gene_expression/allgenes/DE_contrast_",contname,".tsv",sep=""), sep="\t", quote=F, col.names = T, row.names = F)

Reviewer: Where is the `contname` set?

pg <- ggplot(DE_all, aes(x=log2FoldChange, y=logpval, text=paste("Gene: ", gene_name, "<br>", "Log2FC: ", formatC(log2FoldChange, digits=2)))) +
        #this goes after alpha=0.5:

>         geom_jitter(alpha=0.5, aes(color=ifelse(abs(DE_all$log2FoldChange) >= params$logFC_threshold &  DE_all$logpval >= -log10(params$pval_threshold), "Differentially expressed genes", "Non-differentially expressed genes"))) +

Reviewer: Nice 🥳  Is that sufficient or do you also need to specify `width=0.2` or so to make the dots shift sideways?

Author: Added `width=0.2` but in many cases probably it will not change much because the plots are often completely crowded

                    mat$gene_name <- NULL
                    mat <- data.matrix(mat)

>                     # Skip the heatmaps for CI tests because especially png() works only very unreliably for the heatmaps for some reason

Author: No, unfortunately sometimes it also just does not work. I.e. `Unable to start png device` is thrown and the pipeline stops.

> "` pipeline [^5], which was written using the nf-core template [@ewels2020nf]. For differential expression analysis, the read quantification data resulting from `",

Reviewer: Is this `[^5]` the citation number?

Reviewer: If so, it would be missing for the other citations like `@love2014...`

Author: yep, it adds a superscript number and also links it to the citation

Author: ah sorry, no, some citations are just added without the reference thing. The reference is only done for links, i.e. [^5] moves the report down to this link: https://github.com/qbic-pipelines/rnadeseq. The other citations like love2014 are just listed without being clickable

Reviewer: Is there a reason why this is not consistent with the clickable references?

    for (db in datasources) {
        database_string <- paste0(database_string, ", `", db, ": ", database_version_info$sources[[db]]$version, "`")
    }
}

cat(paste0(
"\n## Pathway enrichment analysis

For pathway analysis, the R packages `gprofiler2 v", version_gprofiler2,
" `, `AnnotationDbi v", version_annotation,
"` and `", name_species, " v", version_annotation,

> "` were employed. ", database_string, ".\n",

Reviewer: ```suggestion
"` were used. ", database_string, ".\n",
```

> "Pathways were classified as enriched for those genes with padj <= ", pval_text, "."

Reviewer: ```suggestion
"Pathways were classified as enriched for genes with padj <= ", pval_text, "."
```

Author: counter proposal: `an adjusted p-value <= pval_test`? I prefer having <= (or alternatively "less than or equal to") in the text so people don't wonder, even if that is an edge case
